include/deal.II/lac/affine_constraints.h

```diff
@@ -2203,6 +2232,25 @@ inline AffineConstraints<number>::AffineConstraints(
 
 
 
+template <typename number>
+inline void
+AffineConstraints<number>::add_constraint(
+  const size_type                                  constrained_dof,
+  const std::vector<std::pair<size_type, number>> &dependencies,
+  const number                                     inhomogeneity)
+{
+  Assert(is_constrained(constrained_dof) == false,
+         ExcMessage("You cannot add a constraint for a degree of freedom "
+                    "that is already constrained."));
+
+  add_line(constrained_dof);
+  for (const auto &dep : dependencies)
+    add_entry(constrained_dof, dep.first, dep.second);
```

Comment:
In light of #16121, I believe this is wrong and should be `add_entries(...)`. I will make a quick test to see if it works, otherwise I have to postpone it because my time is constrained.

Reply:
I have a follow-up patch that avoids the repeated calls. Hold on with your timings.